β  Added --beta flag to swig-publish

Meesayen opened the pull request:
This will make it super easy to publish beta versions of modules.

```
swig publish -m <module> --beta
```

It will automatically append `-beta.1` to the version number (unless it finds an already published 
beta package for the current version, in that case it will increment the beta number), and publish
the package with the `beta` dist tag.

RoryH commented:
Previously we've always used tagged versions for testing pre-prod version of modules. I assume that's what this proposed change would allow us do... Was there a discussion around this? I'd like to be able to understand the process around using beta versions before we commit to it. Might need a FECK agenda item to discuss?

Meesayen commented:
It's just a standard-ish substitute for the current way of pre-prod publishing. Currently we can manually specify a version with a `-<something>` appended to it, and the swig publish will create a `test` tag out of it.
I just removed the need for this manual work, used the standardize `beta` tag, and added some check to incrementally advance beta tags for a same version.
